### PR TITLE
Storages refactor

### PR DESCRIPTION
**Pull request by anna1901**
# What was done?

The logic from gigantic MeasurementsStreamStorage was moved to smaller storages responsible for different functionalities.

https://trello.com/c/xgVLTsSE

### CODE REVIEW

File: AirCasting/CoreData/Storages/SyncingMeasurementsStorage.swift [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    /// within a block passed to this methood.
+    /// This ensures thread-safety by dispatching all calls to the queue owned by the NSManagedObjectContext.
+    func accessStorage(_ task: @escaping(HiddenSyncingMeasurementsStorage) -> Void) {
+        context.perform {
+            task(self.hiddenStorage)
+            try? self.hiddenStorage.save()
+        }
+    }
+}
+
```
Comment (jankrzempek) [inline AirCasting/CoreData/Storages/SyncingMeasurementsStorage.swift:36]: Can be deleted

**Review by jankrzempek — APPROVED** [review]
Nice separation was done here 👏
I added few comments and one here regarding whole concept: what would you say to use some kind of `HiddenDefaultStorage` protocol which will have `func save()`. After that every `hiddenStorage` can conform to this protocol, as it seems to be a default thing.

File: AirCasting/CoreData/Storages/SDSyncMeasurementsStorage.swift [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    }
+    
+    func removeDuplicatedMeasurements(for sessionUUID: SessionUUID) throws {
+        let sessionEntity = try context.existingSession(uuid: sessionUUID)
+        sessionEntity.allStreams.forEach({ stream in
+            guard let measurements = stream.allMeasurements else { return }
+            let sortedMeasurements = measurements.sorted(by: { $0.time < $1.time })
+            for (i, measurement) in sortedMeasurements.enumerated() {
+                if i > 0 {
+                    if measurement.time.roundedDownToSecond == sortedMeasurements[i-1].time.roundedDownToSecond {
```
[the remark this answers is not in the document]
Comment (pawelgil) [reply AirCasting/CoreData/Storages/SDSyncMeasurementsStorage.swift:58]: `i` is a convention for naming indicies

**Review by pawelgil — APPROVED** [review]
tanks for doing that

**Pull request merged**